docs/code/getting-started.md

```diff
@@ -675,4 +675,141 @@ Let’s recap what we learned in this section:
 
 Now that you understand more about Gtk, Grids, and using Buttons to alter the properties of other widgets, try packing other kinds of widgets into a window like a Toolbar and changing other properties of [Labels](http://valadoc.org/#!api=gtk+-3.0/Gtk.Label) like `width_chars` and `ellipsize`. Don’t forget to play around with the attach method and widgets that span across multiple rows and columns. Remember that Valadoc is super helpful for learning more about the methods and properties associated with widgets.
 
+# Integrate with Pantheon Online Accounts {#pantheon-online-accounts}
+
+Every application dealing with data will have to connect to several service providers. Let's say that you are a developer of an Address Book application and that you want to integrate with Google Contacts. (In real-life, developers should use the folks library instead to retrieve contacts).
+Most services allows you to only use “tokens” and thus not having to deal with the user password. Token are like passwords as they allow you to interact with the service (here add, modify and remove contacts) but unlike passwords, you are restricted to a specific area of the whole service (you can’t access Google Drive documents with a token made for Contacts) and in top of that, tokens are revocable. The Pantheon Online Account service is a central place to retrieve, store and refresh tokens when they are needed so you don’t have to re-implement all the Authentication procedure. All you need to know is to use the API of the provider of your choice.
+
+## Create a Provider {#poa-create-provider}
+
+There are several default providers shipped with elementary, and much more available in the AppCenter. If the provider you want is already available, jump to the next part as you *MUST* use it. Otherwise, here is how to create a new one.
+
+The Pantheon Online Accounts service has already the major authentication protocols built-in, it is possible to create yourself everything to support a new protocol but we won't cover this here.
+Each service has a corresponding authentication method and mechanism (if the authentication isn't standard, those are often equal), here is a non-exhaustive list of the most used ones:
+
+|   Methods  |                Mechanisms                |
+|  --------  | ---------------------------------------- |
+|  `oauth1`  | `PLAINTEXT` or `HMAC-SHA1` or `RSA-SHA1` |
+|  `oauth2`  |       `web_server` or `user_agent`       |
+| `password` |                `password`                |
+
+To create a provider, you have to ship a .provider file in "/usr/share/accounts/providers/", it is better to name the file as `%provider.provider` (here it’s `google.provider` for example)
```

Comment:
I think this needs more explanation. Where to I ship this file with my app? In the "data" directory? How do I install it to /usr/share/accounts/providers? What do I need to do to my CMake or with packaging?